pkg/autoscaling/handler.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoscaling
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/pd/v4/server"
+	"github.com/unrolled/render"
+)
+
+// Strategy within a HTTP request provides rules and resources to help make decision for auto scaling.
+type Strategy struct {
+	Rules     []*Rule     `json:"rules"`
+	Resources []*Resource `json:"resources"`
+}
+
+// Rule is a set of constraints for a kind of component.
+type Rule struct {
+	Component               string       `json:"component"`
+	CPURule                 *CPURule     `json:"cpu_rule"`
+	StorageRule             *StorageRule `json:"storage_rule"`
+	ScaleOutIntervalSeconds uint64       `json:"scale_out_interval_seconds"`
+	ScaleInIntervalSeconds  uint64       `json:"scale_in_interval_seconds"`
```

Comment:
What about set these attributes as pointer and omitempty, If the user didn't set the interval seconds, we could set a default value instead of 0.

Reply:
From the semantics of the variable, it's not appropriate to use a pointer here. I think we can use another way to adjust the default value.